src/pe_reports/pages.py

```diff
@@ -44,107 +53,134 @@ def overview(prs):
+            df_domains = pd.read_csv("src/pe_reports/data/csv/dhs_domains.csv")
+            run.text = str(int(df_domains.iloc[0]["count"]))
+            font = run.font
+            Paragraph.text_style_ov_val(prs, font)
+        except FileNotFoundError as not_found:
+            print(not_found, "There is no domain data.")
 
         # Overview Value - Active malware associations
         ov_val_03 = Paragraph.text_frame_ov_val(prs, slide, shape, "TextBox 86")
         frame = ov_val_03.paragraphs[0]
         run = frame.add_run()
-        df_malware = pd.read_csv("src/pe_reports/data/csv/dhs_malware.csv")
-        run.text = str(int(df_malware.iloc[0]["count"]))
-        font = run.font
-        font = Paragraph.text_style_ov_val(prs, font)
+        try:
+            df_malware = pd.read_csv("src/pe_reports/data/csv/dhs_malware.csv")
+            run.text = str(int(df_malware.iloc[0]["count"]))
+            font = run.font
+            Paragraph.text_style_ov_val(prs, font)
+        except FileNotFoundError as not_found:
+            print(not_found, "There is no malware data.")
 
         # Overview Value - Web and dark web mentions
         ov_val_04 = Paragraph.text_frame_ov_val(prs, slide, shape, "TextBox 87")
         frame = ov_val_04.paragraphs[0]
         run = frame.add_run()
-        df_vulns = pd.read_csv("src/pe_reports/data/csv/dhs_vulns.csv")
-        run.text = str(int(df_vulns.iloc[0]["count"]))
-        font = run.font
-        font = Paragraph.text_style_ov_val(prs, font)
+        try:
+            df_vulns = pd.read_csv("src/pe_reports/data/csv/dhs_vulns.csv")
+            run.text = str(int(df_vulns.iloc[0]["count"]))
+            font = run.font
+            Paragraph.text_style_ov_val(prs, font)
+        except FileNotFoundError as not_found:
+            print(not_found, "There is no vulnerability data.")
 
         # Overview Value - Credentials exposed in recent posts
         ov_val_05 = Paragraph.text_frame_ov_val(prs, slide, shape, "TextBox 90")
         frame = ov_val_05.paragraphs[0]
         run = frame.add_run()
-        df_web = pd.read_csv("src/pe_reports/data/csv/dhs_web.csv")
-        run.text = str(int(df_web.iloc[0]["count"]))
-        font = run.font
-        font = Paragraph.text_style_ov_val(prs, font)
+        try:
+            df_web = pd.read_csv("src/pe_reports/data/csv/dhs_web.csv")
+            run.text = str(int(df_web.iloc[0]["count"]))
+            font = run.font
+            Paragraph.text_style_ov_val(prs, font)
+        except FileNotFoundError as not_found:
+            print(not_found, "There is no web data.")
 
         # Bar Graph - Top Level Domains used for masquerading
         chart = CategoryChartData()
-        tld_df = pd.read_csv("src/pe_reports/data/csv/dhs_tld_df.csv").fillna(0)
-
-        chart.categories = list(tld_df.columns.values)
-        chart.add_series("Top Level Domains", list(tld_df.loc[0]))
-        x, y, cx, cy = Inches(3.5), Inches(2.9), Inches(2), Inches(1.5)
```

Comment:
The hard-coded values that appear here and later in the code seem fragile.  Can they be computed somehow, or at a minimum stored in constant variables at the top of the file?

Reply:
@jsf9k the `x, y, cx, cy` list as follows:

- Inches(3.5), Inches(2.9), Inches(2), Inches(1.5)
- Inches(5.75), Inches(2.9), Inches(2), Inches(1.5)
- Inches(3.5), Inches(5), Inches(4.8), Inches(2.15)
- Inches(8.25), Inches(2.9), Inches(4.5), Inches(2.0)
- Inches(8.25), Inches(4.9), Inches(4.5), Inches(2.0)

Each has a slight variance to the inputs currently. Also this portion may be expanded or changed on next iteration when a full pe-report build occurs. At that time if we discover patterns in the graph positioning, we can look at constants. Looking to let this one pass for now on the PR.

Reply:
Resolved in 4c928bd

Reply:
@schmelz-ctr, can you create an issue to remind us to revisit the hard-coded `x`, `y`, `cx`, and `cy` later?  I'll resolve this conversation once (1) the issue is created and (2) a comment referencing the issue is added to this file at each location where these values are defined.

Reply:
@schmelz-ctr created #9 in response to this comment.